internal/gengapic/genrest.go

```diff
@@ -324,6 +386,16 @@ func (g *generator) emptyUnaryRESTCall(servName string, m *descriptor.MethodDesc
 	p("    return err")
 	p("}")
 	p("")
+	p("// Set the headers")
+	p("for k, v := range c.xGoogMetadata {")
+	p("  httpReq.Header[k] = v")
+	p("}")
+	// The content type is separate from x-goog metadata.
+	// It's a little more verbose to set here explicitly, but
+	// it's conceptually cleaner.
+	// TODO(dovs) add field headers.
+	p(`httpReq.Header["content-type"] = []string{"application/json",}`)
```

Comment:
nit: Suggest making the field name "canonical" as per the http package doc: Capitalized-Kebab-Case. Below as well.